app/models/manager_refresh/inventory/middleware_manager.rb

```diff
@@ -0,0 +1,62 @@
+module ManagerRefresh::Inventory::MiddlewareManager
+  extend ActiveSupport::Concern
+  include Core
+
+  COMMON_ATTRIBUTES = %i(name ems_ref nativeid properties feed).freeze
+
+  class_methods do
+    def has_middleware_manager_domains(options = {})
```

Comment:
> personally I'd choose something else

@cfcosta I'm all open for better alternatives. It's been discussed [here](https://github.com/ManageIQ/manageiq/pull/14022/files#r103664411) - but we are still in the early stages of implementations, so we could still tweak it...